core/server/worker/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -120,7 +120,7 @@ void createBlockRemote(long sessionId, long blockId, String tierAlias, long init
       throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException;
 
   /**
-   * Frees space to make a specific amount of bytes available in the tier.
+   * Frees space to make a specific amount of bytes available in a best-effort way in the tier.
```

Comment:
Should this still throw `WorkerOutOfSpaceException`?

Reply:
I think it still shows `WorkerOutOfSpaceException` if no space can be freed at all.

Reply:
Does that make sense given that the method is best effort?

Reply:
I think so. If no single byte can be freed, then it is a `worker-out-of-space` situation. Also, at code level, throwing an exception is natural when this happens, otherwise we need some special handling logic.

Reply:
It seems like an arbitrary distinction that not being able to free any bytes is out of space whereas being able to free 1 instead of 10 bytes is fine. We could change the contract to `free space frees at least 1 byte` to make it more reasonable why we throw an exception for 0 bytes freed.